### PR TITLE
Small fixes for cinder driver

### PR DESCRIPTION
**Pull request by MatMaul**

### CODE REVIEW

File: pkg/volume/cinder/attacher.go [hunk 1]
```diff
@@ -124,7 +124,18 @@ func (attacher *cinderDiskAttacher) VolumesAreAttached(specs []*volume.Spec, nod
 		volumesAttachedCheck[spec] = true
 		volumeSpecMap[volumeSource.VolumeID] = spec
 	}
-	attachedResult, err := attacher.cinderProvider.DisksAreAttached(volumeIDList, string(nodeName))
+	instances, res := attacher.cinderProvider.Instances()
+	if !res {
+		return volumesAttachedCheck, fmt.Errorf("failed to list openstack instances")
```
Comment (zhouhaibing089) [inline pkg/volume/cinder/attacher.go]: `Instances()` is a call to get the instance api from cloud provider, the error message should be more appropriate to be `the underlying cloud provider does not claim to support Instances api`.

Comment (MatMaul) [inline pkg/volume/cinder/attacher.go]: Indeed, this is copy/paste from the Attach method so I'll fix the comment there too.

File: pkg/volume/cinder/attacher.go [hunk 2]
```diff
@@ -124,7 +124,18 @@ func (attacher *cinderDiskAttacher) VolumesAreAttached(specs []*volume.Spec, nod
...
-	attachedResult, err := attacher.cinderProvider.DisksAreAttached(volumeIDList, string(nodeName))
+	instances, res := attacher.cinderProvider.Instances()
+	if !res {
+		return volumesAttachedCheck, fmt.Errorf("underlying cloud provider does not claim to support Instances api")
+	}
+	instanceid, err := instances.InstanceID(nodeName)
+	if err != nil {
+		return volumesAttachedCheck, err
+	}
+	if ind := strings.LastIndex(instanceid, "/"); ind >= 0 {
```
Comment (jsafrane) [inline pkg/volume/cinder/attacher.go]: How is `nodeName` different to `instanceId`? Why would `instanceId` contain slash?

@k8s-sig-openstack-pr-reviews, PTAL.

**Comment by jsafrane** [comment]
Looking at the rest of the attacher.go, the same approach is in Attach() and Detach(), so it should be probably the same in VolumesAreAttached().

**Comment by jsafrane** [comment]
@k8s-bot ok to test

File: pkg/volume/cinder/attacher.go (hunk 2 above)
Comment (MatMaul) [inline pkg/volume/cinder/attacher.go]: nodeName is the nova name (used as hostname), instanceId is its uuid. For the / I think it is indeed useless. Like you said I basically copy/paste the working code from Attach ;)

**Comment by MatMaul** [comment]
Thanks for the review, I am still waiting for the CLA for now.

**Comment by xsgordon** [comment]
@idvoretskyi can you add the sig/openstack label?

**Comment by MatMaul** [comment]
My CLA is finally done but it has been fix by someone else :) Let's close it.

**Pull request closed without merging**